crypto/fipsmodule/ec/ec_test.cc

```diff
@@ -1855,19 +1887,63 @@ TEST_P(ECCurveTest, EncodeInfinity) {
                              nullptr, 0, nullptr));
 }
 
-static std::vector<int> AllCurves() {
+TEST_P(ECCurveTest, ECGroupConvForm) {
+  bssl::UniquePtr<BIGNUM> one(BN_new());
+  ASSERT_TRUE(one);
+  ASSERT_TRUE(BN_set_word(one.get(), 1));
+
+  // Ruby depends on |EC_GROUP| to save the used compression format, so we
+  // replicate that scenario. This won't work with our default static curves.
+  bssl::UniquePtr<EC_GROUP> group2(EC_GROUP_dup(group()));
+  EC_GROUP_set_point_conversion_form(group2.get(), POINT_CONVERSION_COMPRESSED);
+
+  // Compute g × 1.
+  bssl::UniquePtr<EC_POINT> point(EC_POINT_new(group()));
+  ASSERT_TRUE(point);
+  ASSERT_TRUE(
+      EC_POINT_mul(group(), point.get(), one.get(), nullptr, nullptr, nullptr));
+
+  // Serialize the points.
+  std::vector<uint8_t> group1_serialized;
+  std::vector<uint8_t> group2_serialized;
+  ASSERT_TRUE(EncodeECPoint(&group1_serialized, group(), point.get(),
+                            EC_GROUP_get_point_conversion_form(group())));
+  ASSERT_TRUE(EncodeECPoint(&group2_serialized, group2.get(), point.get(),
+                            EC_GROUP_get_point_conversion_form(group2.get())));
+
+  if (GetParam().mutable_group) {
+    EXPECT_NE(Bytes(group1_serialized), Bytes(group2_serialized));
+  } else {
+    // |EC_GROUP_set_point_conversion_form| is a no-op when using our default
+    // static groups.
+    EXPECT_EQ(Bytes(group1_serialized), Bytes(group2_serialized));
+  }
+}
+
+static std::vector<CurveParam> AllCurves() {
   const size_t num_curves = EC_get_builtin_curves(nullptr, 0);
   std::vector<EC_builtin_curve> curves(num_curves);
   EC_get_builtin_curves(curves.data(), num_curves);
-  std::vector<int> nids;
-  for (const auto& curve : curves) {
-    nids.push_back(curve.nid);
+  std::vector<CurveParam> nids;
+  for (const auto &curve : curves) {
+    CurveParam curve_param{};
```

Comment:
GCC 4 [doesn't seem to like](https://us-west-2.codebuild.aws.amazon.com/project/eyJlbmNyeXB0ZWREYXRhIjoiWk5IUWJGRGxBcTJ2Mkp4WGF3dnBwYjc5V0ZZYSt5SVVGbkwvODkydTNTaVQ2V2FMN3hwa0tjSWNFemw2QWtCWW5welFWV3lpRFpKVitwejgvelhpRWh3NDNqcWhKalpPYW9hL2tLMDlJSDFPT1NkNyIsIml2UGFyYW1ldGVyU3BlYyI6Ilp4VXRNQXFGM1BZYVlaRkIiLCJtYXRlcmlhbFNldFNlcmlhbCI6MX0%3D/build/fd193587-2582-4ae9-93cc-765b786fd5e2) lack of initialization here.